### PR TITLE
refactor: Deprecation of RMRK

### LINKED ISSUE #10643: Deprecation of RMRK
### Task

1. Remove logic related to rmrk and ksm prefixes

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Refactoring

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10643

- [x] keep teleport \ transfer \ migrate

- [x] search endpoint: `https://polysearch.w.kodadot.xyz/search`
 context: https://github.com/kodadot/polysearch/blob/main/supabase/functions/search/index.ts
solution: Database will be wiped and reindexed on worker side

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by vikiival** [comment]
> search endpoint:

Can you please expand this idea?

Polysearch is in Workers btw

**Comment by Jarsen136** [comment]
> > search endpoint:
> 
> Can you please expand this idea?
> 
> Polysearch is in Workers btw

`Polysearch` search endpoint would return nfts in any chain including rmrk\ksm, which should not be shown on the website.
For now, I add the chain params so it would only search nft from the current supported chain.

If we want to keep mutilple chain search, there are two solutions on the worker side.
1. delete ksm\rmrk items in the database table.
2. manually exclude items from `ksm\rmrk` chain when executing the search SQL

**Comment by vikiival** [comment]
> delete ksm\rmrk items in the database table.

This. Database will be wiped and reindexed again 🩷

**Comment by Jarsen136** [comment]
> > delete ksm\rmrk items in the database table.
> 
> This. Database will be wiped and reindexed again 🩷

Okay, then could you please let me know when you remove it? I will change back the search params on the frontend.

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
As agreed on metahours, RMRK v1 will be deprecated instantly.
RMRK 2 will be decomissioned after discussion

**Comment by Jarsen136** [comment]
> As agreed on metahours, RMRK v1 will be deprecated instantly. RMRK 2 will be decomissioned after discussion

So what should I do to continue with this PR? Should RMRK2 be deprecated? @vikiival

**Comment by vikiival** [comment]
> o what should I do to continue with this PR? Should RMRK2 be deprecated

yes 
- discussion was already opened for more than one month - https://github.com/kodadot/nft-gallery/discussions/10811

We can safely deprecate both

File: components/create/CreateNft.vue [hunk 1]
```diff
@@ -564,25 +539,6 @@ const autoTeleportActions = computed<AutoTeleportAction[]>(() => {
     },
   ]
 
-  if (needsListing.value) {
```
Comment (vikiival) [inline components/create/CreateNft.vue:567]: this disables listing after minting for all no?

File: components/search/SearchSuggestion.vue [hunk 2]
```diff
@@ -311,7 +288,6 @@ import { NeoIcon, NeoSkeleton, NeoTabItem, NeoTabs } from '@kodadot1/brick'
 import { useTopCollections } from '../landing/topCollections/utils/useTopCollections'
 import { fetchCollectionSuggestion } from './utils/collectionSearch'
 import type { DefaultCollectionSuggestion, SearchQuery } from './types'
-import { denyList } from '@/utils/constants'
```
Comment (vikiival) [inline components/search/SearchSuggestion.vue:314]: deny list is used on other platforms too

File: components/stats/StatsHeader.vue [hunk 3]
```diff
@@ -1,26 +0,0 @@
...
-      <h1 class="title is-2">
-        {{ title }}
-      </h1>
-      <p class="subtitle text-xl">
-        {{ subtitle }}
-      </p>
-    </div>
-    <div class="flex-1">
-      <img
-        src="/rmrk-logo-pink-faded.png"
```
Comment (vikiival) [inline components/stats/StatsHeader.vue:13]: remove logo too?

**Review by vikiival — COMMENTED** [review]
Otherwise ok

File: components/create/CreateNft.vue (hunk 1 above)
Comment (Jarsen136) [reply components/create/CreateNft.vue:567]: It's only for Remark

image

File: components/stats/StatsHeader.vue (hunk 3 above)
Comment (Jarsen136) [reply components/stats/StatsHeader.vue:13]: removed

File: components/search/SearchSuggestion.vue (hunk 2 above)
Comment (Jarsen136) [reply components/search/SearchSuggestion.vue:314]: yup, updated

File: components/create/CreateNft.vue (hunk 1 above)
Comment (vikiival) [reply components/create/CreateNft.vue:567]: SO you won't be able to list nfts created on ahp/ ahk?

**Comment by vikiival** [comment]
@kodadot/internal-dev can I please ask you for one more check ?

**Comment by vikiival** [comment]
cc @Jarsen136 you have some conflicts here

File: components/create/CreateNft.vue (hunk 1 above)
Comment (Jarsen136) [reply components/create/CreateNft.vue:567]: We can still create nft and list it simultaneously on ahp/ahk. It was handled somewhere else.
The deleted part of codes in the PR is only for rmrk/ksm network.

image

**Comment by Jarsen136** [comment]
> cc @Jarsen136 you have some conflicts here

done

**Comment by vikiival** [comment]
Tests are failing sir

**Comment by Jarsen136** [comment]
> Tests are failing sir

The failed tests are related to https://github.com/kodadot/nft-gallery/pull/10702 https://github.com/kodadot/nft-gallery/actions/runs/10880638038/job/30187898925

**Comment by vikiival** [comment]
PLZ @hassnian 
@preschian

File: components/carousel/utils/useCarouselEvents.ts [hunk 4]
```diff
@@ -19,7 +18,7 @@ const nftEventVariables = {
 }
 
 const fetchLatestEvents = (chain, type, where = {}, limit = 20) => {
-  const query = chain === 'ksm' ? latestEventsRmrkv2 : latestEvents
+  const query = latestEvents
 
   return useQuery(
     query,
```
Comment (hassnian) [inline components/carousel/utils/useCarouselEvents.ts:24]:
```suggestion
  return useQuery(
    query: latestEvents,
```

File: components/items/ItemsGrid/useItemsGrid.ts [hunk 5]
```diff
@@ -142,7 +131,7 @@ export function useFetchSearch({
     }
 
     // Query path and variables
-    const queryPath = getQueryPath(client.value)
+    const queryPath = client.value
```
Comment (hassnian) [inline components/items/ItemsGrid/useItemsGrid.ts:134]: same here use client where it's needed

File: composables/massmint/useMassMint.ts [hunk 6]
```diff
@@ -40,17 +39,13 @@ export const useCollectionForMint = () => {
...
-    ksm: 'chain-rmrk',
-  }
 
   const doFetch = async () => {
     if (!isLogIn.value) {
       return
     }
 
-    const prefix = queryPath[urlPrefix.value] || urlPrefix.value
+    const prefix = urlPrefix.value
```
Comment (hassnian) [inline composables/massmint/useMassMint.ts:48]: same , no need for a new variable

**Review by hassnian — APPROVED** [review]
otherwise LGTM

**Comment by preschian** [comment]
> * [x]  keep teleport \ transfer \ migrate

@vikiival @Jarsen136 is it safe to delete migrate page/components? or, we already discuss this somewhere before?

**Comment by Jarsen136** [comment]
> > * [x]   keep teleport \ transfer \ migrate
> 
> @vikiival @Jarsen136 is it safe to delete migrate page/components? or, we already discuss this somewhere before?

Here is the context about migrate https://github.com/kodadot/nft-gallery/pull/10667#discussion_r1689594234

File: components/items/ItemsGrid/useItemsGrid.ts (hunk 5 above)
Comment (Jarsen136) [reply components/items/ItemsGrid/useItemsGrid.ts]: updated

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
@JustLuuuu @exezbcz can you please test?

**Comment by JustLuuuu** [comment]
wfm

**Comment by exezbcz** [comment]
works, minted drop without any issues

good job

**Comment by vikiival** [comment]
Thanks 🥰

**Comment by Jarsen136** [comment]
@vikiival Could we merge it?

**Pull request merged**